Realm/Tests/KVOTests.mm

```diff
@@ -300,6 +326,12 @@ bool empty() const {
     } \
 } while (false)
 
+#define AssertSetChanged(s) do { \
+    (r).refresh(); \
+    r.pop_front(); \
```

Comment:
```suggestion
#define AssertSetChanged() do { \
    AssertNotification(r); \
```

This currently is just asserting that there's at most one notification rather than exactly one.

Reply:
Making this change reveals that many of the tests are incorrect.

Reply:
Changed, I see some issues were due to expecting changes for objects with no KVO recorder setup